### PR TITLE
feat(meeting): add new conference resource support

### LINKED ISSUE #2213: [Meeting] No resources to manage cloud conferences
Hi team,

Can we support a resource to create the conference using HuaweiCloud [API](https://support.huaweicloud.com/api-meeting/meeting_21_0014.html).

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
New resource support.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #2213

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add a new conference resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/meeting' TESTARGS='-run=TestAccConference_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/meeting -v -run=TestAccConference_basic -timeout 360m -parallel 4
=== RUN   TestAccConference_basic
=== PAUSE TestAccConference_basic
=== CONT  TestAccConference_basic
--- PASS: TestAccConference_basic (18.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/meeting   18.292s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/meeting' TESTARGS='-run=TestAccConference'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/meeting -v -run=TestAccConference -timeout 360m -parallel 4
=== RUN   TestAccConference_basic
=== PAUSE TestAccConference_basic
=== RUN   TestAccConference_cyclical
=== PAUSE TestAccConference_cyclical
=== RUN   TestAccConference_pwdAuth
=== PAUSE TestAccConference_pwdAuth
=== CONT  TestAccConference_basic
=== CONT  TestAccConference_pwdAuth
=== CONT  TestAccConference_cyclical
--- PASS: TestAccConference_cyclical (25.44s)
--- PASS: TestAccConference_pwdAuth (26.81s)
--- PASS: TestAccConference_basic (42.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/meeting   42.776s
```

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 1]
```diff
@@ -0,0 +1,1111 @@
...
+
+	_, conferenceId, err := analyseIdString(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	opt := buildConferenceGetOpts(conferenceId, d.Get("user_id").(string), token)
+	log.Printf("[DEBUG][Before Update] The GetOpts of the cloud conference is: %v", opt)
+	client := GetMeetingV1Client(conf)
+	resp, err := conferences.Get(client, opt)
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: unnecessary, pls remove

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 2]
```diff
@@ -0,0 +1,1111 @@
...
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"app_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"app_key"},
+				AtLeastOneOf: []string{"account_name"},
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: should use ExactlyOneOf

File: docs/resources/meeting_conferences.md [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+    is_send_calendar = true
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `account_name` - (Optional, String, ForceNew) Specifies the (HUAWEI Cloud meeting) user account name to which the
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md:134]: add a description of the limitation between `account_name` and `app_id`

File: docs/resources/meeting_conferences.md [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+  **0**: Viewable/downloadable via link.
+  **1**: Enterprise users can watch/download.
+  **2**: Attendees can watch/download.
+
+  Only availabe if `record_type` is **2** or **3**.
+
+* `participant_number` - (Optional, Int) Specifies the number of parties in the conference, the maximum number of
+  participants in the conference.
+  **0**: Unlimited.
+  **Greater than 0**: the maximum number of participants in the conference.
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: this is a little confusing, how about just use `defaults to: 0 (Unlimited)`

File: docs/resources/meeting_conferences.md [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+
+* `app_id` - (Optional, String, ForceNew) Specifies the ID of the Third-party application.
+  Changing this parameter will create a new resource.
+
+  -> You can apply for an application and obtain the App ID and App Key in the console.
+
+* `app_key` - (Optional, String, ForceNew) Specifies the Key information of the Third-party APP.
+  Required if `app_id` is set. Changing this parameter will create a new resource.
+
+* `corp_id` - (Optional, String, ForceNew) Specifies the corporation name to which the meeting initiator belongs.
```
Comment (hmbig2) [inline docs/resources/meeting_conferences.md]: s/corporation name/corporation ID

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 6]
```diff
@@ -0,0 +1,1111 @@
...
+				Computed: true,
+				ValidateFunc: validation.IntInSlice([]int{
+					int(RecordAuthTypeLink), int(RecordAuthTypeEnterprise), int(RecordAuthTypeAttendee),
+				}),
+			},
+			"participant_number": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"participant": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: all parameters in this structure are optional, is it ok if the user input an empty structure?

File: docs/resources/meeting_conferences.md [hunk 7]
```diff
@@ -0,0 +1,526 @@
...
+* `email` - (Optional, String) Specifies the email address.
+
+* `sms` - (Optional, String) Specifies the mobile number for SMS notification, maximum of 32 characters.
+
+  -> At least one of `phone`, `email` and `sms` must be set.
+
+<a name="conference_cycle_params"></a>
+The `cycle_params` block supports:
+
+* `cycle` - (Optional, String) Specifies the period type. The valid values are as follows:
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: it's required in the code

File: docs/resources/meeting_conferences.md [hunk 8]
```diff
@@ -0,0 +1,526 @@
...
+  + **Day**
+  + **Week**
+  + **Month**
+
+* `pre_remind` - (Optional, Int) Specifies the number of days for advance conference notice.
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: it's required in the code

File: docs/resources/meeting_conferences.md [hunk 9]
```diff
@@ -0,0 +1,526 @@
...
+  The valid value is range from `0` to `30`, defaults to `1`.
+
+* `start_date` - (Optional, String) Specifies the start date of the recurring conference.
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: it's required in the code

File: docs/resources/meeting_conferences.md [hunk 10]
```diff
@@ -0,0 +1,526 @@
...
+* `end_date` - (Optional, String) Specifies the end date of the recurring conference.
+  The format is `YYYY-MM-DD`.
+
+* `interval` - (Optional, Int) Specifies the cycle interval.
+  For different `cycle` types, the value range of interval are as follows:
+  + **Day**: Means that it will be held every few days, and the valid value is range from `1` to `15`.
+  + **Week**: Means that it is held every few weeks, and the valid value is range from `1` to `5`.
+  + **Month**: Means every few months, the value range is `1` to `3`.
+
+* `points` - (Optional, String) Specifies the conference point in the cycle. Only valid by **Week** and **Month**.
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: the type is List in the code

File: docs/resources/meeting_conferences.md [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+
+* `prolong_time` - (Optional, Int) Specifies the Automatically extend duration (recommended value range is 0-60).
+
+* `waiting_room_enabled` - (Optional, Bool) Specifies whether to open the waiting room (only valid for RTC enterprises).
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID and consist of conference UUID and conference ID.
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: s/and consist/which consists

File: docs/resources/meeting_conferences.md [hunk 12]
```diff
@@ -0,0 +1,526 @@
...
+  + **0**: All users.
+  + **2**: Users within the enterprise.
+  + **3**: The invited user.
+
+* `allow_guest_start` - (Optional, Bool) Specifies whether to allow guests to start conferences (only valid for random
+  ID conferences).
+
+* `guest_password` - (Optional, String) Specifies the guest password (pure number 4-16 digits long).
+
+* `prolong_time` - (Optional, Int) Specifies the Automatically extend duration (recommended value range is 0-60).
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: it's limited to 0-60 in the code, not recommended

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 13]
```diff
@@ -0,0 +1,1111 @@
...
+						"points": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+						},
+					},
+				},
+			},
+			"configuration": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: all parameters in this structure are optional, is it ok if the user input an empty structure?

File: docs/resources/meeting_conferences.md [hunk 14]
```diff
@@ -0,0 +1,526 @@
...
+variable "meeting_room_id" {}
+variable "account_id" {}
+
+resource "huaweicloud_meeting_conference" "test" {
+  app_id  = var.app_id
+  app_key = var.app_key
+
+  topic           = var.conference_topic
+  meeting_room_id = var.meeting_room_id
+  start_time      = "2006-01-02 15:04"
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md]: what will happen if we use an expired time as start time?

File: docs/resources/meeting_conferences.md [hunk 15]
```diff
@@ -0,0 +1,526 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `account_name` - (Optional, String, ForceNew) Specifies the (HUAWEI Cloud meeting) user account name to which the
+  meeting initiator belongs. Changing this parameter will create a new resource.
+
+* `account_password` - (Optional, String, ForceNew) Specifies the user password to which the meeting initiator belongs.
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md]: get rid of "to which the meeting initiator belongs" for password.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: That‘s no problem, the empty structure will be a nil pointer.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 13 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: ditto

File: docs/resources/meeting_conferences.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: good idea

File: docs/resources/meeting_conferences.md [hunk 16]
```diff
@@ -0,0 +1,526 @@
...
+  Only availabe if `app_id` is set. Changing this parameter will create a new resource.
+
+-> Exactly one of account authorization and application authorization you must select.
+
+* `topic` - (Required, String) Specifies the conference topic. The topic can contain `1` to `128` characters.
+
+* `meeting_room_id` - (Required, String) Specifies the cloud meeting room ID.
+
+* `duration` - (Optional, Int) Specifies the duration of the conference.
+  The valid value is range from `15` to `1440`, defaults to `30`.
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md:164]: in minutes?

File: docs/resources/meeting_conferences.md [hunk 17]
```diff
@@ -0,0 +1,526 @@
...
+
+* `start_time` - (Optional, String) Specifies the conference start time (UTC time).
+  The time format is `YYYY-MM-DD hh:mm`, e.g. `2006-01-02 15:04`.
+
+  -> If you want to start a conference at `08:00` (UTC+8), you need to specified the time with `00:00`.
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md]: s/specified/specify

File: docs/resources/meeting_conferences.md [hunk 18]
```diff
@@ -0,0 +1,526 @@
...
+  + **HDVideo**: High-definition video (mutually exclusive with Video, if Video and HDVideo are selected at the same
+    time, the system will select Video by default).
+  + **Data**: Multimedia (If omitted, the system configuration will determines whether to automatically add **Data**).
+
+* `is_auto_record` - (Optional, Int) Specifies whether the conference automatically starts recording, it only takes
+  effect when the recording type is:
+  + **1**: Automatically start recording.
+  + **0**: Do not start recording automatically.
+
+  The default is **0** (not to start automatically).
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md]: "Defaults to **0**" or "The default value is **0**"

File: docs/resources/meeting_conferences.md [hunk 19]
```diff
@@ -0,0 +1,526 @@
...
+  For an aperiodic conference, if the conference notification is sent through a third-party system, this field does not
+  need to be filled in.
+
+* `record_type` - (Optional, Int) Specifies the recording type.
+  + **0**: Disabled.
+  + **1**: Live broadcast.
+  + **2**: Record and broadcast.
+  + **3**: Live + Recording.
+
+  Default is **0** (disabled).
```
Comment (niuzhenguo) [inline docs/resources/meeting_conferences.md]: Defaults to

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: docs/resources/meeting_conferences.md (hunk 5 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 7 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 8 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 10 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 11 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 12 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 15 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 16 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md:164]: sure, the description has been updated.

File: docs/resources/meeting_conferences.md (hunk 17 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 18 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: docs/resources/meeting_conferences.md (hunk 19 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 20]
```diff
@@ -0,0 +1,1145 @@
...
+	if len(resp) < 1 {
+		return diag.Diagnostics{
+			{
+				Severity: diag.Warning,
+				Summary:  "Internal Service Error",
+				Detail: "The corresponding cloud conference is not found after the creation is complete, please " +
+					"contact customer service to solve this problem.",
+			},
+		}
+	}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: return an error here is better

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 21]
```diff
@@ -0,0 +1,1145 @@
...
+	log.Printf("[DEBUG] The result of the subconference is %#v", result)
+	return result
+}
+
+func flattenConferenceCycleParams(cycleParams conferences.CycleParams) []map[string]interface{} {
+	if reflect.DeepEqual(cycleParams, conferences.CycleParams{}) {
+		return nil
+	}
+
+	// No password return, it is a sensitive parameter.
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: remove this comment, there no password in this structure

File: docs/resources/meeting_conferences.md [hunk 22]
```diff
@@ -0,0 +1,527 @@
...
+Import a conference and authenticated by account.
+
+```
+$ terraform import huaweicloud_meeting_conference.test &ltconference_uuid&gt/&ltconference_id&gt/&ltaccount_name&gt/&ltaccount_password&gt
+```
+
+Import a conference and authenticated by `APP ID`/`APP Key`.
+
+```
+$ terraform import huaweicloud_meeting_conference.test &ltconference_uuid&gt/&ltconference_id&gt/&ltapp_id&gt/&ltapp_key&gt/&ltcorp_id&gt/&ltuser_id&gt
```
Comment (Jason-Zhang9309) [inline docs/resources/meeting_conferences.md]: addd a note here as the corp_id and user_id can be empty

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 23]
```diff
@@ -0,0 +1,1145 @@
...
+		if len(resp) < 1 {
+			return diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Internal Service Error",
+					Detail: "The corresponding cloud conference is not found after the creation is complete, please " +
+						"contact customer service to solve this problem.",
+				},
+			}
+		}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: I think we don't need to check the resp here

File: docs/resources/meeting_conferences.md (hunk 14 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: The API will throw an error after we send request.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 23 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: docs/resources/meeting_conferences.md (hunk 22 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 21 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: sure, I has been finished

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 20 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/meeting' TESTARGS='-run=TestAccConference'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/meeting -v -run=TestAccConference -timeout 360m -parallel 4
=== RUN   TestAccConference_basic
=== PAUSE TestAccConference_basic
=== RUN   TestAccConference_cyclical
=== PAUSE TestAccConference_cyclical
=== RUN   TestAccConference_pwdAuth
=== PAUSE TestAccConference_pwdAuth
=== CONT  TestAccConference_basic
=== CONT  TestAccConference_pwdAuth
=== CONT  TestAccConference_cyclical
--- PASS: TestAccConference_cyclical (23.18s)
--- PASS: TestAccConference_pwdAuth (24.40s)
--- PASS: TestAccConference_basic (40.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/meeting   40.139s
```

File: docs/resources/meeting_conferences.md (hunk 9 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 24]
```diff
@@ -0,0 +1,1134 @@
...
+	}
+
+	opt := buildConferenceCreateOpts(d, token)
+	log.Printf("[DEBUG] The createOpts of the cloud conference is: %v", opt)
+	resp, err := conferences.Create(GetMeetingV1Client(conf), opt)
+	if err != nil {
+		return diag.Errorf("error creating cloud conference: %s", err)
+	}
+	if len(resp) < 1 {
+		return diag.Diagnostics{
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: seem to use diag.Errorf is simpler

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 25]
```diff
@@ -0,0 +1,1134 @@
...
+		return nil
+	}
+
+	result := make([]conferences.Participant, len(participants))
+	for i, val := range participants {
+		participant := val.(map[string]interface{})
+		result[i] = conferences.Participant{
+			UserUUID:     participant["user_id"].(string),
+			AccountId:    participant["account_id"].(string),
+			AppId:        participant["name"].(string),
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: wrong field setting, it's name
and in L751, you get the name from `participant.Name`

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 26]
```diff
@@ -0,0 +1,1134 @@
...
+		return nil
+	}
+
+	result := make(map[string]interface{})
+	for _, pwd := range passwords {
+		// Only set passwords of chair role and guest role.
+		if pwd.ConferenceRole == string(ConferenceRoleChair) {
+			result["host"] = pwd.Password
+		}
+		if pwd.ConferenceRole == string(ConferenceRoleChair) {
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: Error? is ConferenceRoleGuest?

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 27]
```diff
@@ -0,0 +1,1134 @@
...
+		opt := buildConferenceUpdateOpts(d, conferenceId, token)
+		_, err = conferences.Update(client, opt)
+		if err != nil {
+			return diag.Errorf("error creating cloud conference: %s", err)
+		}
+		return resourceConferenceRead(ctx, d, meta)
+	default:
+	}
+
+	return diag.Diagnostics{
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: maybe move this return to default block and move L057 and L1064 to here is better

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go [hunk 28]
```diff
@@ -0,0 +1,1134 @@
...
+	}
+	opt := buildConferenceDeleteOpts(d, conferenceId, token)
+	log.Printf("[DEBUG] The DeleteOpts of the cloud conference is: %v", opt)
+	if conferences.Delete(GetMeetingV1Client(conf), opt) != nil {
+		if _, ok := parseConferenceNotFoundError(err).(golangsdk.ErrDefault404); !ok {
+			return diag.Errorf("error deleting conference (%s): %s", conferenceId, err)
+		}
+	}
+
+	d.SetId("")
```
Comment (hmbig2) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: unnecessary, pls remove it

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 27 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 26 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 28 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 25 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go (hunk 24 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/meeting' TESTARGS='-run=TestAccConference'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/meeting -v -run=TestAccConference -timeout 360m -parallel 4
=== RUN   TestAccConference_basic
=== PAUSE TestAccConference_basic
=== RUN   TestAccConference_cyclical
=== PAUSE TestAccConference_cyclical
=== RUN   TestAccConference_pwdAuth
=== PAUSE TestAccConference_pwdAuth
=== CONT  TestAccConference_basic
=== CONT  TestAccConference_pwdAuth
=== CONT  TestAccConference_cyclical
--- PASS: TestAccConference_cyclical (24.34s)
--- PASS: TestAccConference_pwdAuth (26.35s)
--- PASS: TestAccConference_basic (39.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/meeting   39.364s
```

File: huaweicloud/services/meeting/config.go [hunk 29]
```diff
@@ -0,0 +1,202 @@
...
+
+func getMeetingEndpoint(endpoints map[string]string, obj string) string {
+	if endpoint, ok := endpoints["meeting"]; ok {
+		return endpoint
+	}
+	return DefaultEndpoint
+}
+
+// GetMeetingV1Client is a method to general a client with ver.1 meeting endpoint.
+func GetMeetingV1Client(conf *config.Config) *golangsdk.ServiceClient {
```
Comment (ShiChangkuo) [inline huaweicloud/services/meeting/config.go]: suggest to rename to NewMeetingV1Client

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conference.go [hunk 30]
```diff
@@ -0,0 +1,1164 @@
...
+	opt := buildConferenceGetOpts(d.Id(), d.Get("user_id").(string), token)
+	log.Printf("[DEBUG][Before Update] The GetOpts of the cloud conference is: %v", opt)
+	client := GetMeetingV1Client(conf)
+
+	switch d.Get("status").(string) {
+	case string(ConferenceStatusUsing):
+		if d.HasChangeExcept("length") {
+			return diag.Diagnostics{
+				{
+					Severity: diag.Warning,
```
Comment (ShiChangkuo) [inline huaweicloud/services/meeting/resource_huaweicloud_meeting_conference.go]: the level should be Error

File: docs/resources/meeting_conferences.md [hunk 31]
```diff
@@ -0,0 +1,551 @@
...
+variable "meeting_room_id" {}
+variable "account_id" {}
+
+resource "huaweicloud_meeting_conference" "test" {
+  app_id  = var.app_id
+  app_key = var.app_key
+
+  topic           = var.conference_topic
+  meeting_room_id = var.meeting_room_id
+  start_time      = "2006-01-02 15:04"
```
Comment (ShiChangkuo) [inline docs/resources/meeting_conferences.md]: suggest to update the year to 2022 or use the [timestamp](https://www.terraform.io/language/functions/timestamp) function in the usage

File: huaweicloud/services/meeting/config.go (hunk 29 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/config.go]: done.

File: huaweicloud/services/meeting/resource_huaweicloud_meeting_conference.go (hunk 30 above)
Comment (Lance52259) [reply huaweicloud/services/meeting/resource_huaweicloud_meeting_conference.go]: done.

File: docs/resources/meeting_conferences.md (hunk 31 above)
Comment (Lance52259) [reply docs/resources/meeting_conferences.md]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/meeting' TESTARGS='-run=TestAccConference'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/meeting -v -run=TestAccConference -timeout 360m -parallel 4
=== RUN   TestAccConference_basic
=== PAUSE TestAccConference_basic
=== RUN   TestAccConference_cyclical
=== PAUSE TestAccConference_cyclical
=== RUN   TestAccConference_pwdAuth
=== PAUSE TestAccConference_pwdAuth
=== CONT  TestAccConference_basic
=== CONT  TestAccConference_pwdAuth
=== CONT  TestAccConference_cyclical
--- PASS: TestAccConference_cyclical (34.47s)
--- PASS: TestAccConference_pwdAuth (34.74s)
--- PASS: TestAccConference_basic (63.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/meeting   64.304s
```

**Pull request merged**